stl/src/filesystem.cpp

```diff
@@ -503,36 +499,124 @@ _Success_(return == __std_win_error::_Success) __std_win_error
+        _Able_to_change_attributes = true;
+    } else if (_Last_error == __std_win_error::_Access_denied) {
+        // change the underlying HANDLE of _Handle
+        // this might not be a good way of doing it (comment to be removed post review)
+        _Last_error = __std_fs_open_handle(&_Handle._Raw, _Target, __std_access_rights::_Delete, _Flags);
+        if (_Last_error != __std_win_error::_Success) {
+            return {false, _Last_error};
+        }
+    } else {
+        return {false, _Translate_not_found_to_success(_Last_error)};
+    }
+
+    // For Windows 10 1809 or later we have this flag -> FILE_DISPOSITION_FLAG_IGNORE_READONLY_ATTRIBUTE 0x10
+    // This flag also deletes read-only files
+    // NOTE: This is currently undocumented in MSDN. Refer WinBase.h for declarations
+
+    // The following bits are set here
+    // FILE_DISPOSITION_FLAG_DELETE, FILE_DISPOSITION_FLAG_POSIX_SEMANTICS,
+    // FILE_DISPOSITION_FLAG_IGNORE_READONLY_ATTRIBUTE
+    _File_disposition_info_ex _Info_ex{0x3 | 0x10};
+    constexpr auto _FileDispositionInfoExClass = static_cast<FILE_INFO_BY_HANDLE_CLASS>(21);
+    if (SetFileInformationByHandle(
+            reinterpret_cast<HANDLE>(_Handle._Get()), _FileDispositionInfoExClass, &_Info_ex, sizeof(_Info_ex))) {
         return {true, __std_win_error::_Success};
     }
 
-    return {false, __std_win_error{GetLastError()}};
+    _Last_error = __std_win_error{GetLastError()};
+    switch (_Last_error) {
+    // Windows versions older than 1809
+    case __std_win_error::_Invalid_parameter:
+    case __std_win_error::_Invalid_function:
+    case __std_win_error::_Not_supported:
+        break;
+    default:
+        // If we fail for reasons other than those mentioned above just return to the caller
+        return {false, _Last_error};
+    }
 
-#undef _SetFileInformationByHandle
+    // code path for versions older than 1809
+    _Last_error = _Set_delete_flag(_Handle._Raw);
+    if (_Last_error == __std_win_error::_Success) {
+        return {true, __std_win_error::_Success};
+    }
+
+    if (_Last_error == __std_win_error::_Access_denied && _Able_to_change_attributes) {
+
+        FILE_BASIC_INFO _Basic_info;
+        if (!GetFileInformationByHandleEx(_Handle._Get(), FileBasicInfo, &_Basic_info, sizeof(_Basic_info))) {
+            return {false, __std_win_error{GetLastError()}};
+        }
+        // check if FILE_ATTRIBUTE_READONLY is set
+        if (_Basic_info.FileAttributes & FILE_ATTRIBUTE_READONLY) {
+            // try to remove FILE_ATTRIBUTE_READONLY
+            _Basic_info.FileAttributes ^= FILE_ATTRIBUTE_READONLY;
+            if (!SetFileInformationByHandle(_Handle._Get(), FileBasicInfo, &_Basic_info, sizeof(_Basic_info))) {
+                return {false, __std_win_error{GetLastError()}};
+            }
+            // removed FILE_ATTRIBUTE_READONLY, now try to set the delete flag again
+            _Last_error = _Set_delete_flag(_Handle._Raw);

```

Comment:
Another occurrence that should be `_Handle._Get()`.

Reply:
fixed